src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/TypeBuilderImpl.cs

```diff
@@ -945,7 +945,49 @@ public override MethodInfo[] GetMethods(BindingFlags bindingAttr)
         }
 
         [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicFields | DynamicallyAccessedMemberTypes.NonPublicFields)]
-        public override FieldInfo? GetField(string name, BindingFlags bindingAttr) => throw new NotSupportedException();
+        public override FieldInfo? GetField(string name, BindingFlags bindingAttr)
+        {
+            ArgumentNullException.ThrowIfNull(name);
+
+            FieldInfo? match = null;
+
+            BindingFlags matchBindingAttr = bindingAttr ^ BindingFlags.DeclaredOnly;
```

Comment:
Is this name correct? XOR is going to set the bits that match to false. So I would assume this variable is really `nonMatchingBindingAttr`?

It is also possible I just don't understand the intent of the logic here. Perhaps a comment or two to clarify what is going on here.

Reply:
It's kind of excluding the flags that not related to the field signature/visibility, so `nonMatchingBindingAttr` is not good name either, I will try think of a new/better name.

Reply:
This was not needed and now removed